src/components/UserContributionForm.vue

```diff
@@ -0,0 +1,130 @@
+<script setup>
+import { ref, computed } from "vue";
+import { defineProps } from "vue";
+import { recordUserContribution } from "../stores/api.js";
+
+const data = defineProps({
+  description: String,
+  inputOptions: Object,
+  missingFieldName: String,
+  toolName: String,
+  taskId: Number,
+  isError: Boolean,
+  getNextTask: Function,
+});
+
+const inputOptionsArray = computed(() => {
+  return data?.inputOptions
+    ? Object.entries(data.inputOptions).map(([key, value]) => ({ key, value }))
+    : [];
+});
+
+const missingFieldValue = ref(null);
+const errorMessage = ref("");
+
+const submit = async () => {
+  const fieldValue = missingFieldValue.value;
+  const contributionRecord = {
+    value: Array.isArray(fieldValue) ? fieldValue.join() : fieldValue,
+    field: data.missingFieldName,
+    tool: data.toolName,
+  };
+
+  try {
+    const res = await recordUserContribution(data.taskId, contributionRecord);
+    alert(res)
```

Comment:
Hmm this looked nice, but instead of this, can we have a success message around the input field and move to the next task after wards.